### ISSUE TITLE
False negative in AbbreviationAsWordInName, it does not give violation for capital letter after `_`

### ISSUE THREAD

**Issue by Zopsss** [checkstyle/checkstyle#15425]
I have read check documentation: https://checkstyle.org/checks/naming/abbreviationaswordinname.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

Detected at: https://github.com/checkstyle/checkstyle/pull/15411#discussion_r1700123555

Rule of Google style https://google.github.io/styleguide/javaguide.html#s5.2.3-method-names

>Underscores may appear in JUnit test method names to separate logical components of the name, with each component written in [lowerCamelCase](https://google.github.io/styleguide/javaguide.html#s5.3-camel-case), for example transferMoney_deductsFromSource. There is no One Correct Way to name test methods

The following type of method name should give violation for Abbreviation in name:

```java
/** some javadoc. */
public class TestingMethodName {
  /** some javadoc. */
  public void testing_Foo() { // should give warning for such types of names
    // some code
  }
}
```

**Output:**

```
$ java -jar .\checkstyle-10.17.0-all.jar -c .\google_checks.xml .\TestingMethodName.java
Starting audit...
Audit done.
```

[AbbreviationAsWordInName](https://checkstyle.org/checks/naming/abbreviationaswordinname.html) is used to check abbreviation in code, this is a defect in the check. It should give violation if there's a capital letter after `_`

---

**Describe what you expect in detail.**

Update the check's implementation and fix this bug, thus it'll get fixed in the google config automatically.

**Comment by romani** [comment]
Not yet approved , need to think on it.

**Comment by romani** [comment]
I don't think this is a problem of AbbreviationAsWordInName. There is no abbreviation in questionable code.

Google style section that triggered this issue will be resolved by https://github.com/checkstyle/checkstyle/issues/15429

I think we just need to close this issue.
@rdiachenko , please confirm.

**Comment by rdiachenko** [comment]
Agree, it is not a part of `AbbreviationAsWordInName` check and should be done via https://checkstyle.org/checks/naming/methodname.html instead.
